packages/flutter/test/material/theme_data_test.dart

```diff
@@ -170,4 +170,138 @@ void main() {
     expect(theme.applyElevationOverlayColor, isTrue);
   });
 
+  testWidgets('ThemeData.copyWith return appropriate value', (WidgetTester tester) async {
+    final ThemeData theme = ThemeData.light();
+    final ThemeData darkTheme = ThemeData.dark();
+
+    final ThemeData themeDataCopy = theme.copyWith(
+      brightness: darkTheme.brightness,
+      primaryColor: darkTheme.primaryColor,
+      primaryColorBrightness: darkTheme.primaryColorBrightness,
+      primaryColorLight: darkTheme.primaryColorLight,
+      primaryColorDark: darkTheme.primaryColorDark,
+      accentColor: darkTheme.accentColor,
+      accentColorBrightness: darkTheme.accentColorBrightness,
+      canvasColor: darkTheme.canvasColor,
+      scaffoldBackgroundColor: darkTheme.scaffoldBackgroundColor,
+      bottomAppBarColor: darkTheme.bottomAppBarColor,
+      cardColor: darkTheme.cardColor,
+      dividerColor: darkTheme.dividerColor,
+      focusColor: darkTheme.focusColor,
+      hoverColor: darkTheme.hoverColor,
+      highlightColor: darkTheme.highlightColor,
+      splashColor: darkTheme.splashColor,
+      splashFactory: darkTheme.splashFactory,
+      selectedRowColor: darkTheme.selectedRowColor,
+      unselectedWidgetColor: darkTheme.unselectedWidgetColor,
+      disabledColor: darkTheme.disabledColor,
+      buttonTheme: darkTheme.buttonTheme,
+      toggleButtonsTheme: darkTheme.toggleButtonsTheme,
+      buttonColor: darkTheme.buttonColor,
+      secondaryHeaderColor: darkTheme.secondaryHeaderColor,
+      textSelectionColor: darkTheme.textSelectionColor,
+      cursorColor: darkTheme.cursorColor,
+      textSelectionHandleColor: darkTheme.textSelectionHandleColor,
+      backgroundColor: darkTheme.backgroundColor,
+      dialogBackgroundColor: darkTheme.dialogBackgroundColor,
+      indicatorColor: darkTheme.indicatorColor,
+      hintColor : darkTheme.hintColor ,
```

Comment:
Similar nit for all arguments following this one
```suggestion
      hintColor: darkTheme.hintColor,
```